sphinx/domains/python.py

```
@@ -331,15 +367,43 @@ def make_xrefs(self, rolename: str, domain: str, target: str,
         split_contnode = bool(contnode and contnode.astext() == target)
 
         results = []
+        literal_depth = 0
+        pending_literal_open = False
         for sub_target in filter(None, sub_targets):
+            token_contnode = contnode
             if split_contnode:
-                contnode = nodes.Text(sub_target)
+                token_contnode = nodes.Text(sub_target)
 
             if delims_re.match(sub_target):
-                results.append(contnode or innernode(sub_target, sub_target))
+                results.append(token_contnode or innernode(sub_target, sub_target))
+                if literal_depth > 0:
+                    literal_depth += sub_target.count('[')
+                    literal_depth -= sub_target.count(']')
+                    if literal_depth < 0:
+                        literal_depth = 0
+                elif pending_literal_open and '[' in sub_target:
+                    literal_depth = sub_target.count('[') or 1
+                    pending_literal_open = False
+                elif pending_literal_open:
+                    pending_literal_open = False
+                continue
+
+            if literal_depth > 0:
+                results.append(token_contnode or nodes.Text(sub_target))
```

[major] The bracket depth bookkeeping here misfires for literal payloads that contain  or . For example,  produces a  of , so  increments  to 2 and it never returns to zero. Everything that follows is then treated as literal text and never routed through , so  will silently drop the  xref. The slice delimiters already arrive as their own tokens (your earlier branch handles them), so we should not be adjusting  based on the text payload. Please drop the  adjustments from this block (or otherwise guard them so they only execute for actual delimiter tokens).}

---

Thanks for catching that! I dropped the bracket counting from the literal-text path so only the delimiter tokens emitted by the splitter adjust `literal_depth`. That leaves payloads like `Literal['[']` as plain text without permanently tripping the state machine.